### ISSUE TITLE
Inability to install on Python 3.7

### ISSUE THREAD

**Issue by rr-** [pyFFTW/pyFFTW#237]
System and Python version:

```console
rr-@tornado:~$ python -V
Python 3.7.0
rr-@tornado:~$ pip -V
pip 18.0 from /usr/lib/python3.7/site-packages/pip (python 3.7)
rr-@tornado:~$ uname -a
Linux tornado 4.17.2-1-ARCH #1 SMP PREEMPT Sat Jun 16 11:08:59 UTC 2018 x86_64 GNU/Linux
```

Installing via `pip` results in following:

```console
rr-@tornado:~$ pip install --user pyfftw
Collecting pyfftw
  Using cached https://files.pythonhosted.org/packages/c2/2e/b25edc6960fc837e915eb1b38e5f0e3013e32e90aff14a1d0f4556b3d145/pyFFTW-0.10.4.tar.gz
Installing collected packages: pyfftw
  Running setup.py install for pyfftw: started
    Running setup.py install for pyfftw: finished with status 'error'
    Complete output from command /usr/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/pip-install-lto0rrce/pyfftw/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-record-yiegfcxa/install-record.txt --single-version-externally-managed --compile --user --prefix=:
    running install
    running build
    running build_py
    creating build
    creating build/lib.linux-x86_64-3.7
    creating build/lib.linux-x86_64-3.7/pyfftw
    copying pyfftw/__init__.py -> build/lib.linux-x86_64-3.7/pyfftw
    copying pyfftw/version.py -> build/lib.linux-x86_64-3.7/pyfftw
    creating build/lib.linux-x86_64-3.7/pyfftw/builders
    copying pyfftw/builders/builders.py -> build/lib.linux-x86_64-3.7/pyfftw/builders
    copying pyfftw/builders/__init__.py -> build/lib.linux-x86_64-3.7/pyfftw/builders
    copying pyfftw/builders/_utils.py -> build/lib.linux-x86_64-3.7/pyfftw/builders
    creating build/lib.linux-x86_64-3.7/pyfftw/interfaces
    copying pyfftw/interfaces/__init__.py -> build/lib.linux-x86_64-3.7/pyfftw/interfaces
    copying pyfftw/interfaces/_utils.py -> build/lib.linux-x86_64-3.7/pyfftw/interfaces
    copying pyfftw/interfaces/cache.py -> build/lib.linux-x86_64-3.7/pyfftw/interfaces
    copying pyfftw/interfaces/numpy_fft.py -> build/lib.linux-x86_64-3.7/pyfftw/interfaces
    copying pyfftw/interfaces/scipy_fftpack.py -> build/lib.linux-x86_64-3.7/pyfftw/interfaces
    running build_ext
    building 'pyfftw.pyfftw' extension
    creating build/temp.linux-x86_64-3.7
    creating build/temp.linux-x86_64-3.7/tmp
    creating build/temp.linux-x86_64-3.7/tmp/pip-install-lto0rrce
    creating build/temp.linux-x86_64-3.7/tmp/pip-install-lto0rrce/pyfftw
    creating build/temp.linux-x86_64-3.7/tmp/pip-install-lto0rrce/pyfftw/pyfftw
    gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -fno-plt -flto -fuse-linker-plugin -ffat-lto-objects -flto-partition=none -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -fno-plt -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -fno-plt -fPIC -I/tmp/pip-install-lto0rrce/pyfftw/include -I/tmp/pip-install-lto0rrce/pyfftw/pyfftw -I/usr/lib/python3.7/site-packages/numpy/core/include -I/usr/include/python3.7m -c /tmp/pip-install-lto0rrce/pyfftw/pyfftw/pyfftw.c -o build/temp.linux-x86_64-3.7/tmp/pip-install-lto0rrce/pyfftw/pyfftw/pyfftw.o
    In file included from /usr/lib/python3.7/site-packages/numpy/core/include/numpy/ndarraytypes.h:1821,
                     from /usr/lib/python3.7/site-packages/numpy/core/include/numpy/ndarrayobject.h:18,
                     from /usr/lib/python3.7/site-packages/numpy/core/include/numpy/arrayobject.h:4,
                     from /tmp/pip-install-lto0rrce/pyfftw/pyfftw/pyfftw.c:268:
    /usr/lib/python3.7/site-packages/numpy/core/include/numpy/npy_1_7_deprecated_api.h:15:2: warning: #warning "Using deprecated NumPy API, disable it by " "#defining NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION" [-Wcpp]
     #warning "Using deprecated NumPy API, disable it by " \
      ^~~~~~~
    /tmp/pip-install-lto0rrce/pyfftw/pyfftw/pyfftw.c: In function ‘__Pyx_ExceptionSave’:
    /tmp/pip-install-lto0rrce/pyfftw/pyfftw/pyfftw.c:19533:21: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_type’; did you mean ‘curexc_type’?
         *type = tstate->exc_type;
                         ^~~~~~~~
                         curexc_type
    /tmp/pip-install-lto0rrce/pyfftw/pyfftw/pyfftw.c:19534:22: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_value’; did you mean ‘curexc_value’?
         *value = tstate->exc_value;
                          ^~~~~~~~~
                          curexc_value
    /tmp/pip-install-lto0rrce/pyfftw/pyfftw/pyfftw.c:19535:19: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_traceback’; did you mean ‘curexc_traceback’?
         *tb = tstate->exc_traceback;
                       ^~~~~~~~~~~~~
                       curexc_traceback
    /tmp/pip-install-lto0rrce/pyfftw/pyfftw/pyfftw.c: In function ‘__Pyx_ExceptionReset’:
    /tmp/pip-install-lto0rrce/pyfftw/pyfftw/pyfftw.c:19547:24: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_type’; did you mean ‘curexc_type’?
         tmp_type = tstate->exc_type;
                            ^~~~~~~~
                            curexc_type
    /tmp/pip-install-lto0rrce/pyfftw/pyfftw/pyfftw.c:19548:25: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_value’; did you mean ‘curexc_value’?
         tmp_value = tstate->exc_value;
                             ^~~~~~~~~
                             curexc_value
    /tmp/pip-install-lto0rrce/pyfftw/pyfftw/pyfftw.c:19549:22: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_traceback’; did you mean ‘curexc_traceback’?
         tmp_tb = tstate->exc_traceback;
                          ^~~~~~~~~~~~~
                          curexc_traceback
    /tmp/pip-install-lto0rrce/pyfftw/pyfftw/pyfftw.c:19550:13: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_type’; did you mean ‘curexc_type’?
         tstate->exc_type = type;
                 ^~~~~~~~
                 curexc_type
    /tmp/pip-install-lto0rrce/pyfftw/pyfftw/pyfftw.c:19551:13: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_value’; did you mean ‘curexc_value’?
         tstate->exc_value = value;
                 ^~~~~~~~~
                 curexc_value
    /tmp/pip-install-lto0rrce/pyfftw/pyfftw/pyfftw.c:19552:13: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_traceback’; did you mean ‘curexc_traceback’?
         tstate->exc_traceback = tb;
                 ^~~~~~~~~~~~~
                 curexc_traceback
    /tmp/pip-install-lto0rrce/pyfftw/pyfftw/pyfftw.c: In function ‘__Pyx_GetException’:
    /tmp/pip-install-lto0rrce/pyfftw/pyfftw/pyfftw.c:19595:24: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_type’; did you mean ‘curexc_type’?
         tmp_type = tstate->exc_type;
                            ^~~~~~~~
                            curexc_type
    /tmp/pip-install-lto0rrce/pyfftw/pyfftw/pyfftw.c:19596:25: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_value’; did you mean ‘curexc_value’?
         tmp_value = tstate->exc_value;
                             ^~~~~~~~~
                             curexc_value
    /tmp/pip-install-lto0rrce/pyfftw/pyfftw/pyfftw.c:19597:22: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_traceback’; did you mean ‘curexc_traceback’?
         tmp_tb = tstate->exc_traceback;
                          ^~~~~~~~~~~~~
                          curexc_traceback
    /tmp/pip-install-lto0rrce/pyfftw/pyfftw/pyfftw.c:19598:13: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_type’; did you mean ‘curexc_type’?
         tstate->exc_type = local_type;
                 ^~~~~~~~
                 curexc_type
    /tmp/pip-install-lto0rrce/pyfftw/pyfftw/pyfftw.c:19599:13: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_value’; did you mean ‘curexc_value’?
         tstate->exc_value = local_value;
                 ^~~~~~~~~
                 curexc_value
    /tmp/pip-install-lto0rrce/pyfftw/pyfftw/pyfftw.c:19600:13: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_traceback’; did you mean ‘curexc_traceback’?
         tstate->exc_traceback = local_tb;
                 ^~~~~~~~~~~~~
                 curexc_traceback
    /tmp/pip-install-lto0rrce/pyfftw/pyfftw/pyfftw.c: In function ‘__Pyx_ExceptionSwap’:
    /tmp/pip-install-lto0rrce/pyfftw/pyfftw/pyfftw.c:19982:24: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_type’; did you mean ‘curexc_type’?
         tmp_type = tstate->exc_type;
                            ^~~~~~~~
                            curexc_type
    /tmp/pip-install-lto0rrce/pyfftw/pyfftw/pyfftw.c:19983:25: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_value’; did you mean ‘curexc_value’?
         tmp_value = tstate->exc_value;
                             ^~~~~~~~~
                             curexc_value
    /tmp/pip-install-lto0rrce/pyfftw/pyfftw/pyfftw.c:19984:22: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_traceback’; did you mean ‘curexc_traceback’?
         tmp_tb = tstate->exc_traceback;
                          ^~~~~~~~~~~~~
                          curexc_traceback
    /tmp/pip-install-lto0rrce/pyfftw/pyfftw/pyfftw.c:19985:13: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_type’; did you mean ‘curexc_type’?
         tstate->exc_type = *type;
                 ^~~~~~~~
                 curexc_type
    /tmp/pip-install-lto0rrce/pyfftw/pyfftw/pyfftw.c:19986:13: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_value’; did you mean ‘curexc_value’?
         tstate->exc_value = *value;
                 ^~~~~~~~~
                 curexc_value
    /tmp/pip-install-lto0rrce/pyfftw/pyfftw/pyfftw.c:19987:13: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘exc_traceback’; did you mean ‘curexc_traceback’?
         tstate->exc_traceback = *tb;
                 ^~~~~~~~~~~~~
                 curexc_traceback
    In file included from /usr/lib/python3.7/site-packages/numpy/core/include/numpy/ndarrayobject.h:27,
                     from /usr/lib/python3.7/site-packages/numpy/core/include/numpy/arrayobject.h:4,
                     from /tmp/pip-install-lto0rrce/pyfftw/pyfftw/pyfftw.c:268:
    At top level:
    /usr/lib/python3.7/site-packages/numpy/core/include/numpy/__multiarray_api.h:1463:1: warning: ‘_import_array’ defined but not used [-Wunused-function]
     _import_array(void)
     ^~~~~~~~~~~~~
    error: command 'gcc' failed with exit status 1
    
    ----------------------------------------
Command "/usr/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/pip-install-lto0rrce/pyfftw/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-record-yiegfcxa/install-record.txt --single-version-externally-managed --compile --user --prefix=" failed with error code 1 in /tmp/pip-install-lto0rrce/pyfftw/
```

**Comment by rr-** [comment]
I don't have an explanation how this may have worked, but rebooting the system has solved the problem.

**Comment by jakirkham** [comment]
Installing Cython 0.27.3+ and rebuilding should generally fix this as it included fixes for Python 3.7.
